labgrid/external/remote_tmpdir.py

```diff
@@ -0,0 +1,53 @@
+"""
+    RemoteTmpdir is a class that maintains a temporary directory on a
+    remote linux system.
+    Ex. when copying a test script or binary to DUT, RemoteTmpdir assists
+    to prevent pollution of the target.
+
+    RemoteTmpdir is to be created in a fixture, which can take care of a
+    testmodule in a subdirectory. Recommended scope is function, to avoid
+    polution between tests.
+
+    See examples/remote_tmpdir
+"""
+
+import os
+
+
+class RemoteTmpdir:
+
+    # scriptdir is a directory relative to pytest root
+    def __init__(self, shell, testdir=None):
+        stdout, stderr, returncode = shell.run("mktemp -d")
+        assert returncode == 0
+        assert len(stdout) == 1
+        # Use path to prefix copied shells
+        # shell.run(tmp.path+"helloworld")
+        self.path = stdout[0] + '/'
+        self.shell = shell
+        self.testdir = testdir
+
+        if self.testdir is not None and os.path.isfile(self.testdir):
+            raise Exception("RemoteTmpdir: {} is not a directory".format(
+                            self.testdir))
+
+    ## Copy a file or contents of directory the created tmdir
+    # putitem   file or directory to copy
+    #           does not create sub directories, but copies all files
+    def put(self, putitem):
```

Comment:
Would be nice to support multiple files directly in `put()`

```python
def put(self, *files):
```

The entire function would then be

```python
    def put(self, *files):
        for path in files:
            # resolve paths relative to basedir
            if not os.path.isabs(path) and self.basedir:
                path = os.path.join(self.basedir, path)
            if os.path.isfile(path):
                remotepath = self.path + os.path.basename(path)
                self.filetransfer.put(path, remotepath)
            else:
                # copy a whole directory
                for filename in os.listdir(path):
                    localpath = os.path.join(path, filename)
                    remotepath = self.path + filename
                    if os.path.isfile(localpath):
                        self.filetransfer.put(localpath, remotepath)
```